hathor/p2p/sync_v2/manager.py

```diff
@@ -98,7 +98,10 @@ def __init__(self, protocol: 'HathorProtocol', reactor: Optional[Reactor] = None
         self.peer_height = 0
 
         # Latest deferred waiting for a reply.
-        self.deferred_by_key: dict[str, Deferred] = {}
+        self._deferred_txs: dict[str, Deferred[BaseTransaction]] = {}
```

Comment:
We could as well use `VertexId` here for the key.

Reply:
Indeed, but the current implementation uses `key = f'{origin}:{tx_id.hex()}'`, where `origin` is always equal to `mempool`:

https://github.com/HathorNetwork/hathor-core/blob/master/hathor/p2p/sync_v2/manager.py#L1003

It looks like we could actually remove this origin argument altogether, right? It would also simplify some of the downstream code.

Reply:
I'd say we can use the dict only for deferreds with `mempool` origin, rather than removing the origin argument from the sync command (if that's what you meant). And if we need to support a different argument (or drop it altogether) it's independent from this dict.

Reply:
Good idea, I think it's better that I do it on a separate PR, what do you think?

Reply:
Sure, but I think it's probably like a 4 lines change that is really in scope with this PR.

Reply:
True, I thought I would have to remove the `origin` from other places. But that can be done separately. I updated the key type to `VertexId` in https://github.com/HathorNetwork/hathor-core/pull/732/commits/a54f623fb8c531e23ce939007ef5d4a7a9cdd558

Reply:
Great, that's exactly what I meant. Removing `origin` is something else entirely, we'd have to revise if it isn't needed for the implementations of the rest of the sync-v2 protocol that will come in the future.